.github/workflows/ci.yml

```diff
@@ -158,6 +159,16 @@ jobs:
           command: build
           args: --target aarch64-unknown-linux-gnu
 
+  clippy_check:
+    runs-on: ubuntu-latest
+    # Make sure CI fails on all warnings, including Clippy lints
+    env:
+      RUSTFLAGS: "-Dwarnings"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Clippy
+        run: cargo clippy --all-targets --all-features
```

Comment:
Can we make this just check a few conservative lints? I don't want to block builds for _all_ clippy warnings.

Reply:
I removed the `-Dwarnings` env variable, which doesn't fail CI on warnings. The default clippy `deny`s seem reasonable. I added the `deny` undocumented unsafety to `src/lib.rs`